src/CommonLib/Processors/DomainTrustProcessor.cs

```diff
@@ -58,33 +56,26 @@ public async IAsyncEnumerable<DomainTrust> EnumerateDomainTrusts(string domain)
 
                 trust.TargetDomainSid = sid;
 
-                if (int.TryParse(entry.GetProperty(LDAPProperties.TrustDirection), out var td))
-                {
-                    trust.TrustDirection = (TrustDirection) td;
-                }
-                else
-                {
+                if (!entry.TryGetIntProperty(LDAPProperties.TrustDirection, out var td)) {
                     _log.LogTrace("Failed to convert trustdirection for target: {Domain}", domain);
                     continue;
                 }
 
-
+                trust.TrustDirection = (TrustDirection) td;
```

Comment:
Are these casts safe?

Reply:
Yes, int enum casts are always safe in c#